File: frontend/src/types/player.ts
export type Role = "werewolf" | "seer" | "doctor" | "villager";

export type PersuasionProfile =
  | "ethos"
  | "pathos"
  | "logos"
  | "authority_socialproof"
  | "reciprocity_liking"
  | "scarcity_commitment"
  | "baseline";

export interface Player {
  id: string;
  game_id: string;
  agent_name: string;
  role: Role;
  persona: string | null;
  persuasion_profile: PersuasionProfile | null;
  is_mayor: boolean;
  eliminated_round: number | null;
  survived: boolean;
  character_image: string | null;
}

File: backend/app/schemas/player.py
from __future__ import annotations

import uuid

from pydantic import BaseModel, ConfigDict


class PlayerResponse(BaseModel):
    """Response schema for a player."""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    game_id: uuid.UUID
    agent_name: str
    role: str
    persona: str | None = None
    persuasion_profile: str | None = None
    is_mayor: bool
    eliminated_round: int | None = None
    survived: bool
    character_image: str | None = None
